pandas/core/indexes/multi.py

```diff
@@ -1858,6 +1871,11 @@ def _lexsort_depth(self) -> int:
         -------
         int
         """
+        if self.sortorder is not None:
+            return self.sortorder
+        return self._codes_lexsort_depth()
+
+    def _codes_lexsort_depth(self) -> int:
```

Comment:
this is *very* confusing, why do you think you need yet another method here? we want to reduce the api

Reply:
Yes, I totally agree - see this comment for why I thought it was necessary: https://github.com/pandas-dev/pandas/pull/38701#discussion_r550780878 :

> 
> the difficulty I'm running into is that in `pandas/core/indexes/multi.py` there's
> ```python
>         if self.sortorder is not None:
>             if self.sortorder > self._lexsort_depth():
>                 raise ValueError(
>                     "Value for sortorder must be inferior or equal to actual "
>                     f"lexsort_depth: sortorder {self.sortorder} "
>                     f"with lexsort_depth {self._lexsort_depth()}"
>                 )
> ```
> so I can't just move `self.sortorder` into `self._lexsort_depth` else this will never raise, as `self._lexsort_depth()` would always just return `self.sortorder` in this snippet.

Would it be less confusing if this was a module-level function? There must be some way of calling only this part of `_lexsort_depth` because it's used in `_verify_integrity` from pandas/core/indexes/multi.py :

https://github.com/pandas-dev/pandas/blob/d642b67569fdd1f3f3784d3ea79708211dd6a70c/pandas/core/indexes/multi.py#L393-L399

Reply:
I've made it a module-level function, is this clearer?